cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1005,6 +1008,71 @@ void reader::impl::decode_page_headers(hostdevice_vector<gpu::ColumnChunkDesc>&
   pages.device_to_host(stream, true);
 }
 
+void snappy_decompress(device_span<gpu_inflate_input_s> comp_in,
+                       device_span<gpu_inflate_status_s> comp_stat,
+                       size_t max_uncomp_page_size,
+                       rmm::cuda_stream_view stream)
+{
+  size_t num_comp_pages = comp_in.size();
+  size_t temp_size;
+
+  CUDF_EXPECTS(nvcompStatus_t::nvcompSuccess == nvcompBatchedSnappyDecompressGetTempSize(
+                                                  num_comp_pages, max_uncomp_page_size, &temp_size),
+               "Unable to get scratch size for snappy decompression");
```

Comment:
I know this isn't the case here, but I'm always nervous about a macro with a side-effect. I'm used to seeing these things fail because the assert macro was compiled out or changed. This isn't critical here, but it's easy to read over the initialization of `temp_size` inside a macro.

```suggestion
  auto const res = nvcompBatchedSnappyDecompressGetTempSize(
                                                  num_comp_pages, max_uncomp_page_size, &temp_size);
  CUDF_EXPECTS(res == nvcompStatus_t::nvcompSuccess,
               "Unable to get scratch size for snappy decompression");
```